### PR TITLE
Writer extension for generating reports in the cucumber json format

### PR DESCRIPTION
**Pull request by abubeck**
See https://www.relishapp.com/cucumber/cucumber/docs/formatters/json-output-formatter

The extension is tested on Jenkins with the cucumber test result plugin:
https://wiki.jenkins-ci.org/display/JENKINS/Cucumber+Test+Result+Plugin

### CODE REVIEW

File: radish/extensions/cucumber_json_writer.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+                "keyword": feature.keyword,
+                "id": str(feature.id),
+                "name": feature.sentence,
+                "line": feature.line,
+                "description": feature_description,
+                "tags": [],
+                "elements": []
+                }
+            for i in range(len(feature.tags)):
+                feature_json["tags"].append({"name": "@"+feature.tags[i].name, "line": feature.line-len(feature.tags)+i})
```
Comment (timofurrer) [inline radish/extensions/cucumber_json_writer.py]: Please place spaces between operators

File: radish/extensions/cucumber_json_writer.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+                    scenario_json["steps"].append(step_json)
+                feature_json["elements"].append(scenario_json)
+            ccjson.append(feature_json)
+
+        with open(world.config.cucumber_json, "w+") as f:
+            content = json.dumps(ccjson, indent=4, sort_keys=True) #etree.tostring(testrun_element, pretty_print=True, xml_declaration=True, encoding="utf-8")
+            try:
+                if not isinstance(content, str):
+                    content = content.decode("utf-8")
+            except Exception:
```
Comment (timofurrer) [inline radish/extensions/cucumber_json_writer.py]: Too general exception catching

File: radish/extensions/cucumber_json_writer.py [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+                    if step.state is Step.State.FAILED:
+                        step_json["result"]["error_message"] = step.failure.reason
+                    if step.state is Step.State.UNTESTED:
+                        step_json["result"]["status"] = "skipped"
+                    scenario_json["steps"].append(step_json)
+                feature_json["elements"].append(scenario_json)
+            ccjson.append(feature_json)
+
+        with open(world.config.cucumber_json, "w+") as f:
+            content = json.dumps(ccjson, indent=4, sort_keys=True) #etree.tostring(testrun_element, pretty_print=True, xml_declaration=True, encoding="utf-8")
```
Comment (timofurrer) [inline radish/extensions/cucumber_json_writer.py]: comment not needed

File: radish/extensions/cucumber_json_writer.py [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+                "type": "feature",
+                "keyword": feature.keyword,
+                "id": str(feature.id),
+                "name": feature.sentence,
+                "line": feature.line,
+                "description": feature_description,
+                "tags": [],
+                "elements": []
+                }
+            for i in range(len(feature.tags)):
```
Comment (timofurrer) [inline radish/extensions/cucumber_json_writer.py]: Please use `enumerate`

**Comment by timofurrer** [comment]
Thanks for the PR :) :beer:

**Comment by abubeck** [comment]
Thanks for the feedback.
The UTF-8/exception block was actually from the bdd_xml_writer. I don't think it is required in the json case. Therefore I removed the whole section.

**Pull request merged**